alf/examples/icm_mountain_car.gin

```diff
@@ -10,23 +10,41 @@ load_with_random_max_episode_steps.max_steps=1200
 suite_gym.load.max_episode_steps=1000
 
 # algorithm config
+observation_spec=@get_observation_spec()
+action_spec=@get_action_spec()
 
-ActorCriticLoss.entropy_regularization=0.1
-ActorDistributionNetwork.activation_fn=@tf.nn.elu
-ValueNetwork.activation_fn=@tf.nn.elu
+actor/ActorDistributionNetwork.input_tensor_spec=%observation_spec
+actor/ActorDistributionNetwork.output_tensor_spec=%action_spec
+actor/ActorDistributionNetwork.fc_layer_params=(256,)
+actor/ActorDistributionNetwork.activation_fn=@tf.nn.elu
+
+value/ValueNetwork.input_tensor_spec=%observation_spec
+value/ValueNetwork.fc_layer_params=(256,)
+value/ValueNetwork.activation_fn=@tf.nn.elu
+
+ac/Adam.learning_rate=5e-5
 
-create_ac_algorithm.use_icm=1
-create_ac_algorithm.actor_fc_layers=(256,)
-create_ac_algorithm.value_fc_layers=(256,)
-create_ac_algorithm.encoding_fc_layers=(200,)
+icm/EncodingNetwork.input_tensor_spec=%observation_spec
+icm/EncodingNetwork.fc_layer_params=(200,)
+icm/EncodingNetwork.activation_fn=@tf.nn.elu
+icm/tf.TensorSpec.shape=(200,)
```

Comment:
Can we just use icm/TensorSpec here?

Reply:
yes,  there is no other class or function named `TensorSpec` , so  a dotted module name prefix is not necessary required